memory.py

```diff
@@ -16,54 +17,77 @@ def _vector_norms(v:torch.Tensor)->torch.Tensor:
     squared_norms = torch.sum(v * v, dim=1, keepdim=True)
     return torch.sqrt(squared_norms + _EPSILON)
 
-class MLP(nn.Module):
-    '''
-    Multi-layer perceptron class
-    '''
-    def __init__(self, input_size:int, hidden_size:int, output_size:int, activation:torch.nn.modules.activation=torch.nn.ReLU()):
-        """ Init function to initialize a multi-layer perceptron
+def _distance(x:torch.Tensor , y:torch.Tensor, type:str='cosine')->torch.Tensor:
+        """ Compute distances (or other similarity scores) between
+        two sets of samples. Adapted from https://github.com/oscarknagg/few-shot/blob/672de83a853cc2d5e9fe304dc100b4a735c10c15/few_shot/utils.py#L45
 
         Args:
-            input_size (int): Input's dimension
-            hidden_size (int): Number of units in the hidden layer
-            output_size (int): Number of output units
-            activation (torch.nn.modules.activation, optional): Activation function of the hidden layer. Defaults to torch.nn.ReLU().
-        """
-        super(MLP, self).__init__()
-        self.fc1 = torch.nn.Linear(input_size, hidden_size)
-        self.activation = activation
-        self.fc2 = torch.nn.Linear(hidden_size, output_size)
+            x (torch.Tensor):  A tensor of shape (a, b) where b is the embedding dimension. In our paper a=1
+            y (torch.Tensor):  A tensor of shape (m, b) where b is the embedding dimension. In our paper m is the number of samples in support set.
+            type (str, optional): Type of distance to use. Defaults to 'cosine'. Possible values: cosine, l2, dot
 
-    def forward(self, x:torch.Tensor)->torch.Tensor:
-        """ Forward call of multi-layer perceptron
-
-        Args:
-            x (torch.Tensor): Input tensor
+        Raises:
+            NameError: if the name of similarity is unknown
 
         Returns:
-            torch.Tensor: the output of multi-layer perceptron.
+            torch.Tensor: A vector contining the distance of each sample in the vector y from vector x
         """
+        if type == 'cosine':
```

Comment:
This is fine currently. For the future, it's better to create a registry of types and check if the type belongs to the registry.